### PR TITLE
Osu/add pmi versions

### PR DESCRIPTION
**Pull request by MatthewLieber**
Adds a variant for users to choose their pmi version and add a None option to the process_managers variant

### CODE REVIEW

File: var/spack/repos/builtin/packages/mvapich/package.py [hunk 1]
```diff
@@ -59,11 +59,18 @@ class Mvapich(AutotoolsPackage):
...
+        default="simple",
+        values=("simple", "pmi2"),
+        multi=False,
+    )
 
     variant(
         "process_managers",
         description="List of the process managers to activate",
-        values=disjoint_sets(("auto",), ("slurm",), ("hydra", "gforker", "remshell"))
+        values=disjoint_sets(("auto",), ("None",), ("slurm",), ("hydra", "gforker", "remshell"))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mvapich/package.py]: Please don't. If you want to have an identifier for an empty set, just remove the line below:
```python
.phoibit_empy_set()
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by natshineman** [comment]
I think there is some confusion here. Our default case is to have this set to auto (this is the behaviour when unset). There is an actual option to configure called `--with-pm=none` that indicates the user is explicitly disabling a process manager. This is a non-standard use case, but needed for sites like LLNL where they plan to run an alternative launcher/resouce manager that is not directly supported in MVAPICH. So "none" here is actually an explicit disable of the process manager, rather than an empty set.

**Comment by alalazo** [comment]
@natshineman Not sure I follow, but my point is more trivial. Let's not use `None` (uppercase first letter, used by this package only) but remove `proihibit_empy_set`. That will make `none` a possible value (used by many other packages). 

Besides, it's not really clear to me what is the difference between "explicitly disabling process managers" and "the set of allowed process managers is empty".

**Comment by natshineman** [comment]
@alalazo we can do that. I think that the difference within the context of building our package is primarily semantic. If you were to build MVAPICH outside of spack you would have to specify `./configure --with-pm=none ..` to get this behaviour. Just running `./configure ..` would give you `--with-pm=auto`.  I think that Matt was trying to avoid confusion with the Python empty set so that a user familiar with this process would not assume that setting it to empty would result in the default value (auto) being used, as it would in the configure script. So while you are correct that the allowed set of process managers is empty, we have typically approached this from the viewpoint of our standard configure system.

However, when in Spack. @MatthewLieber please make that change and add the logic to convert an empty set from spack to the `--with-pm=none` option in configure.

**Review by alalazo — APPROVED** [review]
Thanks!

**Pull request merged**